x-pack/plugins/logs_shared/public/components/log_ai_assistant/log_ai_assistant.tsx

```diff
@@ -80,7 +87,20 @@ export function LogAIAssistant({ aiAssistant, doc }: LogAIAssistantProps) {
       ) : null}
     </EuiFlexGroup>
   );
-}
+});
 
 // eslint-disable-next-line import/no-default-export
 export default LogAIAssistant;
+
+function withProviders(Component: React.FunctionComponent<LogAIAssistantProps>) {
```

Comment:
This looks neat, but i personally feel this makes the code less Readable.

In the `index.tsx` file we could have simply done

```
export function createLogAIAssistant({
  observabilityAIAssistant: aiAssistant,
}: LogAIAssistantFactoryDeps): LogAIAssistantComponent {
  return ({ observabilityAIAssistant = aiAssistant, ...props }) => (
  <ObservabilityAIAssistantProvider value={observabilityAIAssistant}>
    <LogAIAssistant observabilityAIAssistant={observabilityAIAssistant} {...props} />
    </ObservabilityAIAssistantProvider>
  );
}

And that would have saved the currying here.

This provider is also not a generic reusable code. Just used in here. wdyt ?

Reply:
Ah you can ignore the above comment, i just saw you are also using this component in the `log_entry_flyout` and its required there